### PR TITLE
storybook@v0.16.0 - Dark theme for Storybook.

### PR DESCRIPTION
**Pull request by DamianMullins**
### Added
- PIE dark theme! This theme was designed during the last hackathon by the UX team.
- Custom PIE logo.
- Custom favicon.
- Set up public directory so that we can reference assets in stories and mdx
- `manager-head.html` so we can set up fonts and layout config.

### Changed
- Updated Storybook npm dependencies.
- Override list styles in `preview-head.html`.

## Screenshots

| Before | After |
| ------ | ----- |
| Screenshot 2020-11-16 at 09 54 11 | Screenshot 2020-11-16 at 09 54 18 |

### CODE REVIEW

File: packages/storybook/config/storybook/jetPieThemeDark.js [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+    colorPrimary: '#fff',
+    colorSecondary: '#2a3846',
+
+    // UI
+    appBg: '#0c151d',
+    appContentBg: '#fff',
+    // appBorderColor: '',
+    // appBorderRadius: 0
+
```
Comment (AlixRichards) [inline packages/storybook/config/storybook/jetPieThemeDark.js:15]: Do you still want these commented declarations?

Comment (DamianMullins) [reply packages/storybook/config/storybook/jetPieThemeDark.js:15]: I've left them in so people know they exist if they need to be changed. The Storybook docs have a lot of content but I found it's quite hard to find information like this.

**Review by AlixRichards — APPROVED** [review]
I've left a small question but aside from that looks good

File: packages/storybook/config/storybook/jetPieThemeDark.js (hunk 1 above)
Comment (AlixRichards) [reply packages/storybook/config/storybook/jetPieThemeDark.js:15]: Cool - just wanted to check

File: packages/storybook/config/storybook/manager-head.html [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+    }
+
+    body {
+        font-feature-settings: "tnum"; /* Enable tabular numbers */
+    }
+
+    /*
+     * Sidebar override styles
+     */
+    div.sidebar-subheading {
```
Comment (DamianMullins) [inline packages/storybook/config/storybook/manager-head.html:63]: **NOTE**: Unfortunately Storybook doesn't provide a nicer way to include custom styles. You _can_ import CSS files into the `manager` and `preview` JS modules and they'll work locally, but once you build the project and deploy the styles won't be there.

File: packages/storybook/config/storybook/manager-head.html [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+            // Awkwardly dump the second stage @font-face blocks in the head
+            var style = document.createElement('style');
+            // Note: Edge supports WOFF2
+            style.innerHTML = '@font-face { font-family: JustEatBasis; font-weight: 600; src: ' + fontUrls + '; }';
+            document.head.appendChild(style);
+            document.documentElement.classList.add('webfonts-loaded');
+        }
+    })();
+    // This sets the initial width of the nav sidebar.
+    localStorage.setItem('storybook-layout', JSON.stringify({
```
Comment (DamianMullins) [inline packages/storybook/config/storybook/manager-head.html:34]: **NOTE**: Storybook doesn't provide a way to set this via config so we have to resort to applying the same behaviour as when you manually change the sidebar size.

**Review by kevinrodrigues — APPROVED** [review]

**Pull request merged**